src/Kernel/Magic/ViewMagic.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        private const string ParameterNameOperationName = "__operationName__";
+
+        /// <summary>
+        ///     Constructs a new magic command given a resolver used to find
+        ///     operations and functions, and a configuration source used to set
+        ///     configuration options.
+        /// </summary>
+        public ViewMagic(ISymbolResolver resolver, IConfigurationSource configurationSource) : base(
+            "view",
+            new Documentation
+            {
+                Summary = "Outputs an HTML-based visualization of the execution path of a given operation."
+            })
+        {
+            this.SymbolResolver = resolver;
+            this.ConfigurationSource = configurationSource;
+        }
+
+        /// <summary>
+        ///      The symbol resolver used by this magic command to find the operation
+        ///      to be visualized.
+        /// </summary>
+        public ISymbolResolver SymbolResolver { get; }
+
+        /// <summary>
+        ///     The configuration source used by this magic command to control
+        ///     visualization options.
+        /// </summary>
+        public IConfigurationSource ConfigurationSource { get; }
+
+        /// <inheritdoc />
+        public override ExecutionResult Run(string input, IChannel channel) =>
+            RunAsync(input, channel).Result;
+
+        /// <summary>
+        ///     Outputs a visualization of an operation given a string with its name and a JSON
+        ///     encoding of its arguments.
+        /// </summary>
+        public async Task<ExecutionResult> RunAsync(string input, IChannel channel)
+        {
+            // Parse input parameters
+            var inputParameters = ParseInputParameters(input, firstParameterInferredName: ParameterNameOperationName);
+
+            var name = inputParameters.DecodeParameter<string>(ParameterNameOperationName);
+            var symbol = SymbolResolver.Resolve(name) as IQSharpSymbol;
+            if (symbol == null) throw new InvalidOperationException($"Invalid operation name: {name}");
+
+            var tracer = new ExecutionPathTracer();
+
+            // Simulate operation and attach `ExecutionPathTracer` to keep track of operations performed
+            // in its execution path
+            using var qsim = new QuantumSimulator()
+                .WithJupyterDisplay(channel, ConfigurationSource)
+                .WithStackTraceDisplay(channel)
+                .WithExecutionPathTracer(tracer);
+            var value = await symbol.Operation.RunAsync(qsim, inputParameters);
+
+            // Retrieve the `ExecutionPath` traced out by the `ExecutionPathTracer`
+            var executionPath = tracer.GetExecutionPath();
+
+            // Render empty div with unique ID as cell output
+            var divId = $"execution-path-container-{Guid.NewGuid().ToString()}";
+            var content = new ExecutionPathVisualizerContent(executionPath.ToJson(), divId);
```

Comment:
Following up on my comment above, the `ToJson()` call here is what causes the double-serialization. Converting to a JToken would allow composing the two JSON serialziations instead of nesting them.